### PR TITLE
Refactor: isolate directives scopes (PARTIAL)

### PR DESCRIPTION
**Pull request by juancoen**
- Changing directives to have an isolated scope so RAML AST
transformations can be done at the controller level on each
different component while keeping the RAML AST to implement new RAML
1.0 features.

### CODE REVIEW

File: src/app/directives/close-button.js [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
       templateUrl: 'directives/close-button.tpl.html',
       replace: true,
-      controller: ['$scope', '$rootScope', function($scope, $rootScope) {
+      scope: {
+        handleCloseClick: '&'
+      },
+      controller: ['$scope', function($scope) {
         $scope.close = function () {
-          var $inactiveElements = jQuery('.raml-console-tab').add('.raml-console-resource').add('li');
+          jQuery('.raml-console-tab.raml-console-is-active')
```
Comment (AlejoFernandez) [inline src/app/directives/close-button.js]: Let's not use jQuery if possible

Comment (AlejoFernandez) [inline src/app/directives/close-button.js]: In general, controllers should not mess with DOM, it is a bad practice.
I know that some of that code was already there, I'm just pointing some issues that will arise in the future.

File: src/app/directives/documentation.js [hunk 2]
```diff
@@ -5,13 +5,45 @@
...
+      scope: {
+        methodInfo: '=',
+        securitySchemes: '='
+      },
       controller: ['$scope', function($scope) {
+        function getResponseInfo() {
+          var responseInfo = {};
+          var responses    = $scope.methodInfo.responses;
+
+          if (responses) {
```
Comment (AlejoFernandez) [inline src/app/directives/documentation.js]: Let's remove indentation by doing things like:
```
if (!responses) {
  return {};
}
```

File: src/app/directives/close-button.js (hunk 1 above)
Comment (juancoen) [inline src/app/directives/close-button.js]: I agree, I'll file issues for all of this problems so they can be fixed as part of another PR

**Pull request merged**